File: Tutorial4-3ComponentBuilders/src/main/java/com/example/tutorial4_3componentbuilders/MyApplication.java
package com.example.tutorial4_3componentbuilders;

import android.app.Application;

import com.example.tutorial4_3componentbuilders.di.AppComponent;
import com.example.tutorial4_3componentbuilders.di.AppComponentWithBindsInstance;
import com.example.tutorial4_3componentbuilders.di.AppModule;
import com.example.tutorial4_3componentbuilders.di.DaggerAppComponent;
import com.example.tutorial4_3componentbuilders.di.DaggerAppComponentWithBindsInstance;

public class MyApplication extends Application {

    private AppComponent appComponent;

    private AppComponentWithBindsInstance appComponentWithBindsInstance;

    @Override
    public void onCreate() {
        super.onCreate();

        appComponent = DaggerAppComponent
                .builder()
                .appModule(new AppModule(this))
                .build();

        appComponentWithBindsInstance = DaggerAppComponentWithBindsInstance
                .builder()
                .application(this)
                .build();
    }

    public AppComponent getAppComponent() {
        return appComponent;
    }

    public AppComponentWithBindsInstance getAppComponentWithBindsInstance() {
        return appComponentWithBindsInstance;
    }
}

File: Tutorial4-3ComponentBuilders/src/main/java/com/example/tutorial4_3componentbuilders/MyApplication.kt
package com.example.tutorial4_3componentbuilders

import android.app.Application
import com.example.tutorial4_3componentbuilders.di.*

class MyApplication : Application() {

    lateinit var appComponent: AppComponent
        private set

    lateinit var appComponentWithBindsInstance: AppComponentWithBindsInstance
        private set

    override fun onCreate() {
        super.onCreate()

        appComponent = DaggerAppComponent
                .builder()
                .appModule(AppModule(this))
                .build()

        appComponentWithBindsInstance = DaggerAppComponentWithBindsInstance
                .builder()
//                .application(this)
                .build()
    }

}